File: cddl/contrib/dtracetoolkit/Code/Ruby/func_slow.rb
#!./ruby -w

def func_c
  print "Function C\n"
  i = 0
  while i < 300000
     i = i + 1
     j = i + 1
  end
end

def func_b
  print "Function B\n"
  i = 0
  while i < 200000
     i = i + 1
     j = i + 1
  end
  func_c
end

def func_a
  print "Function A\n"
  i = 0
  while i < 100000
     i = i + 1
     j = i + 1
  end
  func_b
end

func_a

File: cddl/contrib/dtracetoolkit/Code/Perl/func_slow.pl
#!./perl -w

sub func_c {
    print "Function C\n";
    for (my $i = 0; $i < 3000000; $i++) { my $j = $i + 1; }
}

sub func_b {
    print "Function B\n";
    for (my $i = 0; $i < 2000000; $i++) { my $j = $i + 1 ; }
    func_c();
}

sub func_a {
    print "Function A\n";
    for (my $i = 0; $i < 1000000; $i++) { my $j = $i + 1; }
    func_b();
}

func_a();
